### PR TITLE
[iOS] Fix push tests

### PR DESCRIPTION
**Pull request by shrishrirang**
- Update the e2e tests on iOS to work with the latest backend code.
- Minor dotnet backend fixes

### CODE REVIEW

File: sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoAppDelegate.m [hunk 1]
```diff
@@ -12,6 +12,10 @@ @implementation ZumoAppDelegate
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
 {
     if([application respondsToSelector:@selector(registerForRemoteNotifications)]) {
+        
+        UIUserNotificationSettings *notificationSettings = [UIUserNotificationSettings settingsForTypes:UIUserNotificationTypeAlert | UIUserNotificationTypeBadge | UIUserNotificationTypeSound categories:nil];
+        [application registerUserNotificationSettings:notificationSettings];
+        
         [application registerForRemoteNotifications];
```
Comment (phvannor) [inline sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoAppDelegate.m]: shouldn't this now happen in the function triggered from registerUserNotificatinSettings being successful?

File: sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoPushTests.m [hunk 2]
```diff
@@ -105,31 +105,13 @@ - (BOOL)compareExpectedPayload:(NSDictionary *)expected withActual:(NSDictionary
...
-    id badge = originalPayload[@"badge"];
-    if (badge) {
-        [aps setValue:badge forKey:@"badge"];
-    }
-    
-    id sound = originalPayload[@"sound"];
-    if (sound) {
-        [aps setValue:sound forKey:@"sound"];
-    }
+- (NSDictionary *)zumoPayloadToApsPayload:(NSDictionary *)payload {
```
Comment (phvannor) [inline sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoPushTests.m]: do we really need a function to do this?  The server is inconsistent in what it sends back?

File: sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoAppDelegate.m (hunk 1 above)
Comment (shrishrirang) [inline sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoAppDelegate.m]: This is as-is from the documentation at https://azure.microsoft.com/en-us/documentation/articles/mobile-engagement-ios-get-started/

File: sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoPushTests.m (hunk 2 above)
Comment (shrishrirang) [inline sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoPushTests.m]: Yes it no more needs to be a separate function. This function is used to transform the request payload. The server is inconsistent in what it expects.

File: sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoAppDelegate.m (hunk 1 above)
Comment (phvannor) [inline sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoAppDelegate.m]: hmm, ok.  I see Apple shows the same pattern, if we make a new revision, maybe add a log in the userFail just for completeness, but ok as is then.

File: sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoPushTests.m (hunk 2 above)
Comment (phvannor) [inline sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoPushTests.m]: Let me know if you plan to update, also can you put a issue up in the repo to fix the API on the server side to be consistent long term.

**Comment by shrishrirang** [comment]
Ping

**Pull request merged**